test/calibration/experiments/test_ramsey_xy.py

```diff
@@ -71,6 +72,27 @@ def test_update_calibrations(self):
         self.assertTrue(len(self.cals.parameters_table(parameters=[freq_name])["data"]), 2)
         self.assertTrue(abs(f01 - (freq_shift + FakeArmonk().defaults().qubit_freq_est[0])) < tol)
 
+    def test_update_with_failed_analysis(self):
+        """Test that calibration update handles analysis producing no results
+
+        Here we test that the experiment does not raise an unexpected exception
+        or hang indefinitely. Since there are no analysis results, we expect
+        that the calibration update will result in an ERROR status.
+        """
+        backend = MockRamseyXY(freq_shift=0)
+
+        class NoResults((BaseAnalysis)):
```

Comment:
```suggestion
        class NoResults(BaseAnalysis):
```
Is this intentional?